hips/hip-9999.md

```diff
@@ -0,0 +1,56 @@
+---
+hip: 9999
+title: "Pre-defined release dates for Helm"
+authors: [ "Marc Khouzam <marc.khouzam@montreal.ca>" ]
+created: "2020-09-04"
+type: "process"
+status: "draft"
+---
+
+## Abstract
+
+Have well-known cyclical release dates for Helm minor and patch releases.
+Have published release dates for Helm major releases, once such a release has entered the implementation phase.
+
+## Motivation
+
+Contributors and users/organisations greatly appreciate knowing in advance when the next release will be available.
+It allows contributors to plan their work and organisations to schedule migration plans; it also avoids disappointing delays and provides users with clear timelines for access to bug fixes and features.
+
+## Rationale
+
+Minor and patch releases for Helm normally occur a couple of times a year.  They therefore lend themselves well to cyclical, pre-defined release dates.
+
+As it is not recommended to use Helm with a version of Kubernetes that is newer than the version Helm was compiled against ([see here][helm-skew]), there is value in aligning the date for Helm minor releases with [Kubernetes releases][kubernetes-dates], which occur every 3 to 4 months.
+
+Helm major releases happen rarely and therefore do not lend themselves to pre-defined, cyclical release dates.  However, once a major release has been defined and has entered the implementation phase, a specific release date should be chosen and announced.
```

Comment:
> and has entered the implementation phase

As we discovered with Helm 3, it was not easy to choose a specific release date for 3.0.0, especially during the implementation phase. There were design issues and regressions we had to work through, all of which caused significant delays in the release schedule (Helm 3 was delayed over a year from the proposed finish date, if I'm not mistaken).

However, it was easier to plan a release date once we reached Helm 3.0.0-beta.1. That's when we started to get a clearer picture of what needed to be done before the final release was cut.

What if we changed "has entered the implementation phase" with "the first beta release has been published"?

Reply:
Good point.  I think for a fixed date, the beta.1 would give a much better chance of respecting that date, which is important, or else it is kind of useless setting a date.

For major releases, do we have a criteria to trigger the work to begin?
I'm asking because we often tell contributors that "this change will need to wait for helm 4", but I don't think we have a good story on when helm 4 could possibly happen.  That is probably another discussion that what this HIP covers though.

I'll update the HIP according to your suggestion.

Reply:
> For major releases, do we have a criteria to trigger the work to begin?

If I can try to word it correctly, it'd be a general agreement from the core maintainers that there's a sizable number of tasks on the table to warrant asking users to migrate to a new major version. There also has to be maintainers willing to put in the work. :)